backend/core/src/listings/listings.service.ts

```diff
@@ -49,62 +49,79 @@ export class ListingsService {
   }
 
   public async list(origin: string, params: ListingsQueryParams): Promise<PaginatedListingsDto> {
-    let qb = this.getQueryBuilder()
+    // Inner query to get the sorted listing ids of the listings to display
+    // TODO(avaleske): Only join the tables we need for the filters that are applied
+    const innerFilteredQuery = this.listingRepository
+      .createQueryBuilder("listings")
+      .select("listings.id", "listings_id")
+      .leftJoin("listings.property", "property")
+      .groupBy("listings.id")
+      .orderBy({ "listings.id": "DESC" })
+
+    const qb = this.getFullyJoinedQueryBuilder()
```

Comment:
Readability nit: maybe put everything related to the fully-joined query (i.e. this line and `qb.orderBy`) down below, just above `qb.andWhere("listings.id IN (" + innerFilteredQuery.getQuery() + ")")`? Seems slightly easier to me to first think only about the inner query, and then to think about how that informs the overall query.

Reply:
Good idea! Fixed.